airflow/ti_deps/deps/trigger_rule_dep.py

```diff
@@ -146,6 +146,11 @@ def _get_relevant_upstream_map_indexes(upstream_id: str) -> int | range | None:
                 expanded_ti_count = _get_expanded_ti_count()
             except (NotFullyPopulated, NotMapped):
                 return None
+            if ti.map_index < 0:
```

Comment:
I wonder why this case does not raise NotFullyPopulated

Reply:
From this https://github.com/apache/airflow/blob/1b26cc05ff553685c50a15f56a954e465ed43478/airflow/ti_deps/deps/trigger_rule_dep.py#L128, it seems like it's the `supposed` number it could expand into but I think this case is because the group has expanded but the task hasn't?

Reply:
I digged around a bit, this counts as expanded because there is indeed enough information to expand the operator, but the other depended task (`t2`) has not finished. This makes me think the fix is not right; we should investigate why `t2` is not considered relevant instead.

The bug is in `_iter_upstream_conditions`, I think. Need more digging.

Reply:
Yeah.

```
(Pdb) p str(or_(*_iter_upstream_conditions(relevant_tasks=upstream_tasks)).compile(session.connection(), compile_kwargs={"literal_binds": True}))
"task_instance.task_id = 't1' OR task_instance.task_id = 'tg1.t2' AND task_instance.map_index < 0 OR task_instance.task_id = 'tg1.t2' AND task_instance.map_index = -1"
```

This is not right; since `t3` does not depend on `t2` for mapping, it should depend on all of its tis, not just -1.